### PR TITLE
RISC-V: implement helper call snippet

### PR DESCRIPTION
**Pull request by janvrany**

### CODE REVIEW

File: compiler/riscv/codegen/RVHelperCallSnippet.cpp [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+      distance = (intptr_t)(_restartLabel->getCodeLocation()) - (intptr_t)cursor;
+      if (VALID_UJTYPE_IMM(distance))
+         {
+         // b distance
+         *(int32_t *)cursor = TR_RISCV_UJTYPE(TR::InstOpCode::_jal, cg()->machine()->getRealRegister(OMR::RealRegister::zero), distance >> 2);
+         cursor += RISCV_INSTRUCTION_LENGTH;
+         }
+      else
+         {
+         TR_ASSERT(false, "Target too far away.  Not supported yet");
```
Comment (0xdaryl) [inline compiler/riscv/codegen/RVHelperCallSnippet.cpp:64]: You should make this a `TR_ASSERT_FATAL`

File: compiler/riscv/codegen/RVHelperCallSnippet.cpp [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      intptr_t restartLocation = (intptr_t)restartLabel->getCodeLocation();
+      if (VALID_UJTYPE_IMM((intptr_t)restartLocation - (intptr_t)bufferPos))
+         {
+         printPrefix(pOutFile, NULL, bufferPos, 4);
+         trfprintf(pOutFile, "jal \tzero, " POINTER_PRINTF_FORMAT "\t\t; Back to ", restartLocation);
+         print(pOutFile, restartLabel);
+         }
+      else
+         {
+         TR_ASSERT(false, "Target too far away.  Not supported yet");
```
Comment (0xdaryl) [inline compiler/riscv/codegen/RVHelperCallSnippet.cpp:105]: You should make this a `TR_ASSERT_FATAL`

File: compiler/riscv/codegen/RVHelperCallSnippet.cpp [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+   printSnippetLabel(pOutFile, snippet->getSnippetLabel(), bufferPos, getName(snippet));
+
+   char *info = "";
+   intptr_t target = (intptr_t)(snippet->getDestination()->getSymbol()->castToMethodSymbol()->getMethodAddress()) ;
+   int32_t distance;
+   if (isBranchToTrampoline(snippet->getDestination(), bufferPos, distance))
+      {
+      target = (intptr_t)distance + (intptr_t)bufferPos;
+      info = " Through trampoline";
+      TR_ASSERT(VALID_UJTYPE_IMM(distance), "Trampoline too far away.");
```
Comment (0xdaryl) [inline compiler/riscv/codegen/RVHelperCallSnippet.cpp:86]: `TR_ASSERT_FATAL`

File: compiler/riscv/codegen/RVHelperCallSnippet.cpp [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+   {
+   uint8_t *cursor = cg()->getBinaryBufferCursor();
+   intptr_t distance = (intptr_t)(getDestination()->getSymbol()->castToMethodSymbol()->getMethodAddress()) - (intptr_t)cursor;
+
+   getSnippetLabel()->setCodeLocation(cursor);
+
+   if (!VALID_UJTYPE_IMM(distance))
+      {
+      distance = TR::CodeCacheManager::instance()->findHelperTrampoline(getDestination()->getReferenceNumber(), (void *)cursor) - (intptr_t)cursor;
+      TR_ASSERT(VALID_UJTYPE_IMM(distance), "Trampoline too far away.");
```
Comment (0xdaryl) [inline compiler/riscv/codegen/RVHelperCallSnippet.cpp:40]: `TR_ASSERT_FATAL`

File: compiler/riscv/CMakeLists.txt [hunk 5]
```diff
@@ -40,6 +40,7 @@ compiler_library(riscv
 	${CMAKE_CURRENT_LIST_DIR}/codegen/OMRTreeEvaluator.cpp
 	${CMAKE_CURRENT_LIST_DIR}/codegen/OpBinary.cpp
 	${CMAKE_CURRENT_LIST_DIR}/codegen/UnaryEvaluator.cpp
+        ${CMAKE_CURRENT_LIST_DIR}/codegen/RVHelperCallSnippet.cpp
```
Comment (0xdaryl) [inline compiler/riscv/CMakeLists.txt:43]: Fix indenting please.

**Review by 0xdaryl — CHANGES_REQUESTED** [review]
Generally looks good.  Minor suggestions only.

**Comment by janvrany** [comment]
Everything mentioned is addressed in 3dab8a0.

**Review by 0xdaryl — APPROVED** [review]

**Pull request merged**